packages/common/src/store/pages/chat/slice.ts

```diff
@@ -355,10 +355,22 @@ const slice = createSlice({
       delete state.optimisticUnreadMessagesCount
     },
     sendMessage: (
-      _state,
-      _action: PayloadAction<{ chatId: string; message: string }>
+      state,
+      action: PayloadAction<{
+        chatId: string
+        message: string
+        resend?: boolean
+        messageId?: string // Required if resend = true
```

Comment:
Do we need these as separate fields? Could we just have`resendMessageId?`?

Reply:
brilliant